source/adapters/level_zero/queue.cpp

```diff
@@ -1818,6 +1850,9 @@ ur_queue_handle_t_::ur_queue_group_t::getZeQueue(uint32_t *QueueGroupOrdinal) {
   if (QueueIndex != 0) {
     ZeCommandQueueDesc.flags = ZE_COMMAND_QUEUE_FLAG_EXPLICIT_ONLY;
   }
+  if (Queue->usingCounterBasedEvents()) {
```

Comment:
this should be other way around, if queue is in order , then you can create counter based events.

Reply:
ZeQueue hasn't been created at this point. If we are using counter-based event then the zequeue needs to reflect that.

Reply:
I believe @MichalMrozek is referring to the SYCL Queue being passed in AKA the UR Queue object in this line. You can check if this Queue was created to be in-order using the Queue->isInOrderQueue(). 

Also I believe it should also be noted that ZE_COMMAND_QUEUE_FLAG_IN_ORDER flag should only used to create immediate command lists specifically using zeCommandListCreateImmediate(), while I see this function is used to create an ordinary queue for regular command lists using zeCommandQueueCreate(). Please correct me if I'm wrong @MichalMrozek 

Reference at: https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=ze_command_queue_flag_in_order#_CPPv4N23ze_command_queue_flag_t30ZE_COMMAND_QUEUE_FLAG_IN_ORDERE

Reply:
@raiyanla Thanks for your input. Queue->usingCounterBasedEvents() does check for whether the Queue is in order with Queue->isInOrderQueue(). It also checks for ZE_COMMAND_QUEUE_FLAG_IN_ORDER, which I believe is also required for zeCommandQueueCreate(). See example in L0 coformance test [here](https://github.com/oneapi-src/level-zero-tests/blob/5990783066d8114e220ffd57061201e4f8d7cb17/conformance_tests/core/test_cmdlist/src/test_cmdlist.cpp#L823)

Reply:
@winstonzhang-intel Gotcha, yeah I just saw your usingCounterBasedEvents() already checks for isInOrderQueue() now. Thanks for linking that L0 conformance test, that is interesting, it seems to conflict with the specification definition of it above, since the spec link indicates its ZE_COMMAND_QUEUE_FLAG_IN_ORDER should be for immediate command lists only. @MichalMrozek Can you confirm which is is appropriate? Since the spec and conformance tests are conflicting.

Reply:
For regular  command lists the flag is ZE_COMMAND_LIST_FLAG_IN_ORDER , passed in command list descriptor
For immediate command lists the flag is ZE_COMMAND_QUEUE_FLAG_IN_ORDER , passed in queue descriptor